r/R/dataset.R

```diff
@@ -43,19 +43,36 @@
 #'   * `NULL` for no partitioning
 #'
 #' The default is to autodetect Hive-style partitions.
+#' @param unify_schemas logical: should all data fragments (files, `Dataset`s)
+#' be scanned in order to create a unified schema from them? If `FALSE`, only
+#' the first fragment will be inspected for its schema. Use this
```

Comment:
I like the logic of this parameter. I think it is better than what we have now in C++ (not needing two parameters to get this behaviour), although the C++ might be fine to have more fine grained control.